### DISCUSSION TITLE
[2022/02/08] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#191] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/190 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
[メルカリShops の技術スタック、その後](https://engineering.mercari.com/blog/entry/20220207-mercari-shops-tech-stack-next/)

**Comment by kobayang** [comment]
[kintone フロントエンドリアーキテクチャプロジェクトで大切にしていること](https://blog.cybozu.io/entry/2022/02/04/171154)

**Comment by kobayang** [comment]
https://headlessui.dev/

コードが綺麗だった

**Reply by kobayang** [reply]
https://github.com/tailwindlabs/headlessui

**Reply by kobayang** [reply]
```tsx
<Menu>
  <Menu.Button>More</Menu.Button>
  <Menu.Items>
    <Menu.Item>
    </Menu.Item>
  </Menu.Items>
</Menu>
```

みたいな、コンポーネントの入れ子構造みたいになっているの結構いいなと思った。
コンポーネントの組み合わせを表現しつつ、1つのsemanticsが1つのコンポーネントに対応していて分かりやすい

**Reply by kobayang** [reply]
wantedly のデザインシステムもこんな感じで表現するといいかもなーとI/Fを見てて思った

```
<Button appearance="primary">
  <Button.TouchArea>
    {({ reaction }) => (
      <Button.Button reaction={reaction} onClick={handleClick} >
        <Button.Text>Hoge</Button.Text>
      </Button.Button>
    )}
  </Button.TouchArea>
</Button>
```

`Button.Button` がダサすぎるけど...
